detect_compiler fails for pure C projects except for MSVC

At the beginning of the function, a check is made for C++ and C compiler presence. The variables `_compiler` and `_compiler_version` are then set based on the outcome.

But then a series of compiler-specific branches follow, where instead of `_compiler_version`, `CMAKE_CXX_COMPILER_VERSION` is used, thus negating the intention of the C/C++ compiler check made in the beginning. See: https://github.com/conan-io/cmake-conan/blob/develop2/conan_provider.cmake#L276
This code fails on pure C projects, leading to the error:

>     CMake Error at conan_provider.cmake:281 (string):
>       string sub-command REPLACE requires at least four arguments.
>     Call Stack (most recent call first):
>       conan_provider.cmake:379 (detect_compiler)
>       conan_provider.cmake:584 (detect_host_profile)
>       CMakeLists.txt:6 (find_package)

This is easy to fix though, just use _compiler_version. For example:

>     elseif(_compiler MATCHES AppleClang)
>         set(_compiler "apple-clang")
>         string(REPLACE "." ";" VERSION_LIST ${CMAKE_CXX_COMPILER_VERSION})
>         list(GET VERSION_LIST 0 _compiler_version)

should be

>     elseif(_compiler MATCHES AppleClang)
>         set(_compiler "apple-clang")
>         string(REPLACE "." ";" VERSION_LIST ${_compiler_version})
>         list(GET VERSION_LIST 0 _compiler_version)

This way I was able to get my pure C project to use Conan via CMake correctly.

Edit:

I should have added an example CMakeLists.txt:

```
cmake_minimum_required(VERSION 3.31)
project(intro2 C)

set(CMAKE_C_STANDARD 23)

find_package(SDL3)

add_executable(intro2 main.c)

target_link_libraries(intro2 sdl::sdl)
```

and conanfile.txt:

```
[requires]
sdl/[>=3.0 <4.0]

[generators]
CMakeDeps
```

--- a/conan_provider.cmake
+++ b/conan_provider.cmake
@@ -275,15 +275,15 @@ function(detect_compiler compiler compiler_version compiler_runtime compiler_run
 
     elseif(_compiler MATCHES AppleClang)
         set(_compiler "apple-clang")
-        string(REPLACE "." ";" VERSION_LIST ${CMAKE_CXX_COMPILER_VERSION})
+        string(REPLACE "." ";" VERSION_LIST ${_compiler_version})
         list(GET VERSION_LIST 0 _compiler_version)
     elseif(_compiler MATCHES Clang)
         set(_compiler "clang")
-        string(REPLACE "." ";" VERSION_LIST ${CMAKE_CXX_COMPILER_VERSION})
+        string(REPLACE "." ";" VERSION_LIST ${_compiler_version})
         list(GET VERSION_LIST 0 _compiler_version)
     elseif(_compiler MATCHES GNU)
         set(_compiler "gcc")
-        string(REPLACE "." ";" VERSION_LIST ${CMAKE_CXX_COMPILER_VERSION})
+        string(REPLACE "." ";" VERSION_LIST ${_compiler_version})
         list(GET VERSION_LIST 0 _compiler_version)
     endif()
 

